css/css-syntax/escaped-eof.html

```diff
@@ -0,0 +1,53 @@
+<!doctype html>
+<title>Escaped EOF</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<meta name="author" title="Tab Atkins-Bittner">
+<link rel=help href="https://drafts.csswg.org/css-syntax/#consume-escaped-code-point">
+<link rel=help href="https://drafts.csswg.org/css-syntax/#consume-string-token">
+<link rel=help href="https://drafts.csswg.org/css-syntax/#consume-token">
+
+<style>foo { --foo:foo\</style>
+<style>foo { --foo:1foo\</style>
+<style>foo { --foo:url(foo\</style>
+<style>foo { --foo:"foo\</style>
+
+<script>
+
+function threw(fn) {
+    try{fn();}
+    catch(e) { return true; }
+    return false;
+}
+
+test(()=>{
+    assert_true(threw(()=>{document.querySelector("#123");}), "numeric hash token is invalid in a selector");
+    assert_false(threw(()=>{document.querySelector("#foo\\");}), "escaped-EOF in a hash token is valid in a selector");
```

Comment:
If you just perform the querySelector, any exceptions will cause automatic test failures, with better failure messages than "got true, expected false".